src/components/WalletModal/Option.tsx

```diff
@@ -62,25 +64,26 @@ const IconWrapper = styled.div`
   `};
 `
 
-type OptionProps = {
-  connection: Connection
-  activate: () => void
-  pendingConnectionType?: ConnectionType
-}
-export default function Option({ connection, pendingConnectionType, activate }: OptionProps) {
-  const isPending = pendingConnectionType === connection.type
+export default function Option({ connection }: { connection: Connection }) {
+  const { activationState, tryActivation } = useActivationState()
+  const closeDrawer = useCloseAccountDrawer()
+  const activate = () => tryActivation(connection, closeDrawer)
+
+  const isSomeConnectorPending = activationState.status === ActivationStatus.PENDING
+  const isCurrentOptionPending = isSomeConnectorPending && activationState.connection.type === connection.type
   const isDarkMode = useIsDarkMode()
-  const content = (
+
+  return (
     <TraceEvent
       events={[BrowserEvent.onClick]}
       name={InterfaceEventName.WALLET_SELECTED}
       properties={{ wallet_type: connection.getName() }}
       element={InterfaceElementName.WALLET_TYPE_OPTION}
     >
       <OptionCardClickable
-        onClick={!pendingConnectionType ? activate : undefined}
-        clickable={!pendingConnectionType}
-        disabled={Boolean(!isPending && !!pendingConnectionType)}
+        onClick={!isSomeConnectorPending ? activate : undefined}
+        clickable={!isSomeConnectorPending}
```

Comment:
Yeah this is definitely overly complicated ty

I refactored the css so that `!clickable` disables pointer events, and then replaced `disabled` w/ a more simple `selected` var